### DISCUSSION TITLE
How to configure embedded infinispan jdbc cache store in Keycloak.X 19

### DISCUSSION THREAD

**Discussion by ghilainm** [keycloak/keycloak#14460] [Archive]
I am trying to configure this in Keycloak cache configuration:

```
<string-keyed-jdbc-store xmlns="urn:infinispan:config:store:jdbc:13.0" dialect="POSTGRES">
                    <connection-pool connection-url="jdbc:postgresql://localhost/keycloak"
                                     username="keycloak"
                                     password="password"
                                     driver="com.mysql.jdbc.Driver"/>
                    <string-keyed-table create-on-start="true"
                                        prefix="ISPN_STRING_TABLE">
                        <id-column name="ID_COLUMN"
                                   type="VARCHAR(255)" />
                        <data-column name="DATA_COLUMN"
                                     type="BINARY" />
                        <timestamp-column name="TIMESTAMP_COLUMN"
                                          type="BIGINT" />
                        <segment-column name="SEGMENT_COLUMN"
                                        type="INT"/>
                    </string-keyed-table>
                </string-keyed-jdbc-store>
```

And I get the following error when starting Keycloak:

keycloak_1      | 2022-09-16 12:53:36,135 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: org.infinispan.commons.CacheConfigurationException: ISPN000327: Cannot find a parser for element 'string-keyed-jdbc-store' in namespace 'urn:infinispan:config:store:jdbc:13.0'. Check that your configuration is up-to date for Infinispan '13.0.9.Final' and if you have the proper dependency in the classpath
keycloak_1      | 2022-09-16 12:53:36,135 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: ISPN000327: Cannot find a parser for element 'string-keyed-jdbc-store' in namespace 'urn:infinispan:config:store:jdbc:13.0'. Check that your configuration is up-to date for Infinispan '13.0.9.Final' and if you have the proper dependency in the classpath

How am I supposed to add the necessary dependency to the Keycloak classpath? I tried copying the lib to the container but that does not work:

COPY infinispan-cachestore-jdbc-13.0.8.Final.jar /opt/keycloak/lib/lib/main

Any guidance on this would be really appreciated.

**Comment by baalimago** [comment]
After much digging I found this: [https://github.com/thomasdarimont/keycloak-project-example/tree/main/deployments/local/clusterx/haproxy-database-ispn ](https://github.com/thomasdarimont/keycloak-project-example/tree/main/deployments/local/clusterx/haproxy-database-ispn )which descibes a solution. 

Specifically look here: [https://github.com/thomasdarimont/keycloak-project-example/blob/main/deployments/local/clusterx/haproxy-database-ispn/docker-compose.yml#L24C14-L24C14)](https://github.com/thomasdarimont/keycloak-project-example/blob/main/deployments/local/clusterx/haproxy-database-ispn/docker-compose.yml#L24C14-L24C14)

**Comment by ghilainm** [comment]
Seems not that straightforward :sweat_smile: !

Thx for the information :). It would be nice if this could be properly documented somewhere.
